src/libraries/System.Transactions.Local/tests/OleTxTests.cs

```diff
@@ -589,11 +594,15 @@ private static void Test(Action action)
                 action();
                 return;
             }
-            catch (TransactionException e) when (e.InnerException is TransactionManagerCommunicationException)
+            catch (Exception e) when (e is TransactionManagerCommunicationException or TransactionException { InnerException: TransactionManagerCommunicationException })
             {
                 if (--nRetries == 0)
                 {
-                    throw;
+                    // We've continuously gotten XACT_E_TMNOTAVAILABLE for the entire retry window - MSDTC is unavailable in some way.
+                    // We don't want this to make our CI flaky, so we swallow the exception and skip all subsequent tests.
+                    s_isTestSuiteDisabled = true;
+
+                    return;
```

Comment:
@stephentoub do we have a standard way to detect CI, so that this can continue throwing when run locally (e.g. GetEnvironmentVariable("CI") or something)?

Reply:
I'm not aware of a method we call anywhere to ask whether we're running in CI, but we do have an attribute we apply to suppress a test when in CI:
```C#
[Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)]
```

Reply:
Thanks, I'll take a look at how that's implemented.

Reply:
Here's the implementation you're looking for:

https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/Discoverers/SkipOnCIDiscoverer.cs#L19-L21

Reply:
Thank you!